internal/infrastructure/kubernetes/proxy/resource_provider_test.go

```diff
@@ -128,6 +202,14 @@ func TestDeployment(t *testing.T) {
 			expected, err := loadDeployment(tc.caseName)
 			require.NoError(t, err)
 
+			sortEnv := func(env []corev1.EnvVar) {
```

Comment:
this shouldnt be needed, since its a list
```
if extensionEnv := deploymentConfig.Container.Env; extensionEnv != nil {
 		env = append(env, extensionEnv...)
 	}
```

Reply:
current logic using `map`, so needs to sort and make sure test data.